core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/ExtensionLoader.java

```diff
@@ -145,8 +141,7 @@ public static void loadVault(ServiceExtensionContext context, ExtensionLoader lo
     public List<InjectionContainer<ServiceExtension>> loadServiceExtensions() {
         List<ServiceExtension> serviceExtensions = loadExtensions(ServiceExtension.class, true);
 
-        //the first sort is only to verify that there are no "upward" dependencies from PRIMORDIAL -> DEFAULT
-        var sips = sortExtensions(serviceExtensions);
+        var sips = new DependencyGraph().of(serviceExtensions);
 
         return Collections.unmodifiableList(sips);
```

Comment:
Not done in this PR, but this `unmodifiableList` wrapper is not needed, as the list returned by `of` is already immutable

Reply:
`DependencyGraph.of` returns a `List`, the fact that it happens to be immutable is an implementation detail that could change at any time, so we cannot rely on it.

Reply:
@paullatzelsperger same is for `loadServiceExtensions` method and the `createExtensions` that calls it, since it's internal code (not to be extended or so) I wouldn't be so defensive

Reply:
mh, you're probably right. will ax it.